src/rdkafka.c

```diff
@@ -346,6 +346,29 @@ rd_kafka_resp_err_t rd_kafka_oauthbearer_set_token_failure(rd_kafka_t *rk,
 #endif
 }
 
+rd_kafka_error_t *rd_kafka_set_sasl_creds(rd_kafka_t *rk, const char *username, const char *password) {
+        char errstr[256];
+        rd_kafka_conf_t *conf = &rk->rk_conf; /* Not using rd_kafka_conf because of const qualifier */
+
+        rwlock_wrlock(&conf->sasl.lock);
```

Comment:
No need for an rwlock, this is not in the fast-path. Use a regular mtx_t (mtx_plain).